File: HDCommon/app/src/main/java/com/haoda/common/net/annotation/Description.java
package com.haoda.common.net.annotation;

import java.lang.annotation.ElementType;
import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;
import java.lang.annotation.Target;

/**
 * @author Isidore
 */
@Target({ElementType.METHOD,ElementType.FIELD,ElementType.TYPE,ElementType.PARAMETER})
@Retention(RetentionPolicy.SOURCE)
public @interface Description {
    String value() default "";
}

File: HDCommon/libKAPT/src/main/java/Description.kt
package org.kotlin.annotationProcessor

import java.lang.annotation.Retention
import java.lang.annotation.RetentionPolicy

/**
 * @author Isidore
 */
@Target(AnnotationTarget.FUNCTION, AnnotationTarget.PROPERTY_GETTER, AnnotationTarget.PROPERTY_SETTER, AnnotationTarget.FIELD, AnnotationTarget.ANNOTATION_CLASS, AnnotationTarget.CLASS, AnnotationTarget.VALUE_PARAMETER)
@Retention(RetentionPolicy.RUNTIME)
annotation class Description(val value: String = "") 